org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -57,10 +61,13 @@
 public class JavaStackTraceHyperlink implements IHyperlink {
 
 	private final TextConsole fConsole;
+	private String generatedLink;
```

Comment:
Looks like member generatedLink can be accessed from different threads (used in UIJob). So it should be at least volatile, even better use a AtomicReference<String> or access only via synchronized setter/getter.

Reply:
will go with AtomicReference 👍